src/Forum/Forum.router.ts

```diff
@@ -78,27 +75,13 @@ export class ForumRouter extends Router {
       if (isTPCollection(collection)) {
         // This logic is repeated in the Collection.router, we should generalize this behavior.
         const items = await Item.findOrderedByCollectionId(collectionId)
-        const slicedItems = items
-          .slice(0, MAX_FORUM_ITEMS)
-          .map((item) => Bridge.toFullItem(item, collection))
 
-        if (collection.forum_id) {
-          const postData = await getPost(collection.forum_id)
-          await updatePost(
-            collection.forum_id,
-            buildCollectionForumUpdateReply(postData.raw, slicedItems)
-          )
-          return
-        } else {
-          const { id: postId, link } = await createPost(
-            buildCollectionForumPost(collection, slicedItems)
-          )
-          await Collection.update<CollectionAttributes>(
-            { forum_link: link, forum_id: postId },
-            { id: collectionId }
-          )
-          return link
-        }
+        return this.service.upsertThirdPartyCollectionForumPost(
```

Comment:
should we wait for this upsert?

Reply:
Yes, we should because of the catch. Thanks